analyzer/javacg-opal/src/test/java/eu/fasten/analyzer/javacgopal/data/JSONUtilsTest.java

```diff
@@ -0,0 +1,157 @@
+package eu.fasten.analyzer.javacgopal.data;
+
+import eu.fasten.analyzer.javacgopal.data.exceptions.OPALException;
+import eu.fasten.core.data.ExtendedRevisionJavaCallGraph;
+import eu.fasten.core.data.JSONUtils;
+import eu.fasten.core.merge.CallGraphMerger;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import org.jooq.tools.csv.CSVReader;
+import org.json.JSONObject;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import org.skyscreamer.jsonassert.JSONAssert;
+import org.skyscreamer.jsonassert.JSONCompareMode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+class JSONUtilsTest {
+    private static final Logger logger = LoggerFactory.getLogger(JSONUtilsTest.class);
+
+    private static ExtendedRevisionJavaCallGraph graph, artifact, dependency;
+    private static List<MavenCoordinate> coords;
+    private int batchVolume = 20; //percentage of batch tests to be executed in the build
+
+    @BeforeAll
+    static void setUp() throws IOException, OPALException {
+
+        var coordinate =
+            new MavenCoordinate("com.github.shoothzj", "java-tool", "3.0.30.RELEASE", "jar");
+        graph = PartialCallGraph.createExtendedRevisionJavaCallGraph(coordinate,
+            "", "CHA", 1574072773);
+
+        coordinate =
+            new MavenCoordinate("abbot", "costello", "1.4.0", "jar");
+        artifact = PartialCallGraph.createExtendedRevisionJavaCallGraph(coordinate,
+            "", "CHA", 1574072773);
+
+        coordinate =
+            new MavenCoordinate("abbot", "abbot", "1.4.0", "jar");
+        dependency = PartialCallGraph.createExtendedRevisionJavaCallGraph(coordinate,
+            "", "CHA", 1574072773);
+
+        CallGraphMerger.mergeCallGraph(artifact, List.of(dependency), "CHA");
+
+        coords =
+            readDataCSV(new File(Objects.requireNonNull(Thread.currentThread().getContextClassLoader()
+            .getResource("121Coordinates.csv")).getFile()));
+    }
+
+    @Test
+    void toJSONString() throws IOException {
+
+        final var ser1 = avgConsumption(graph, "direct", "direct", 20, 20);
+        final var ser2 = avgConsumption(graph, "jsonObject", "jsonObject", 20, 20);
+        JSONAssert.assertEquals(ser1, ser2, JSONCompareMode.LENIENT);
```

Comment:
I would rather use `JSONCompareMode.STRICT` or `JSONCompareMode.NON_EXTENSIBLE` because we want to have exactly the same JSON fields with the two implementation (see http://jsonassert.skyscreamer.org/apidocs/org/skyscreamer/jsonassert/JSONCompareMode.html)

Reply:
fixed de5c7d9